relay-sampling/src/config.rs

```diff
@@ -36,6 +36,13 @@ impl SamplingConfig {
     pub fn unsupported(&self) -> bool {
         !self.rules_v2.iter().all(SamplingRule::supported)
     }
+
+    /// Filters the sampling rules by the given [`RuleType`].
+    pub fn filter_rules(&self, rule_type: RuleType) -> impl Iterator<Item = &SamplingRule> {
```

Comment:
@jan: moved from `ProjectState` because it was simply calling the SamplingConfig anyway, and it makes more sense to have it here based on where im calling it

Reply:
indeed, it was used by `fn is_trace_fully_sampled`, which took in a state as an argument. I changed the signature to take in a `SamplingConfig` because the function only used state to extract the config